src/com/seafile/seadroid2/cameraupload/MediaObserverService.java

```diff
@@ -0,0 +1,134 @@
+            // necessary.
+
+            switch (key) {
+
+                // if video upload has been switched on, do a full sync, to upload
+                // any older videos already recorded.
+                case SettingsManager.CAMERA_UPLOAD_ALLOW_VIDEOS_SWITCH_KEY:
+                    if (settingsManager.isVideosUploadAllowed())
+                        doFullResync = true;
+                    break;
+
+                // same goes for if the list of selected buckets has been changed
+                case SettingsManager.SHARED_PREF_CAMERA_UPLOAD_BUCKETS:
+                    doFullResync = true;
+                    break;
+
+                // the repo changed, also do a full resync
+                case SettingsManager.SHARED_PREF_CAMERA_UPLOAD_REPO_ID:
+                    doFullResync = true;
+                    break;
+            }
+
+            if (cameraManager.isCameraUploadEnabled() && doFullResync) {
+                Log.i(DEBUG_TAG, "Doing a full resync of all media content.");
+                cameraManager.performFullSync();
+            }
+        }
+    };
+
+    @Override
+    public void onCreate() {
+        Log.d(DEBUG_TAG, "onCreate");
+
+        settingsManager.registerSharedPreferencesListener(settingsListener);
+        cameraManager = new CameraUploadManager(getApplicationContext());
+        registerContentOberservers();
+
+        if (cameraManager.isCameraUploadEnabled()) {
+            // do a sync in case we missed something while we weren't observing
+            cameraManager.performSync();
+        }
+    }
+
+    @Override
+    public void onDestroy() {
+        Log.d(DEBUG_TAG, "onDestroy");
+        settingsManager.unregisterSharedPreferencesListener(settingsListener);
+        unregisterContentOberservers();
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        return START_STICKY;
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    private void registerContentOberservers() {
+        mediaObserver = new MediaObserver();
+
+        getApplicationContext().getContentResolver().registerContentObserver(
+                MediaStore.Images.Media.EXTERNAL_CONTENT_URI, false, mediaObserver);
+
+        getApplicationContext().getContentResolver().registerContentObserver
+                (MediaStore.Video.Media.EXTERNAL_CONTENT_URI, false, mediaObserver);
+
+        Log.i(DEBUG_TAG, "Started watchting for new media content.");
+    }
+
+    private void unregisterContentOberservers() {
+        this.getApplicationContext().getContentResolver()
+                .unregisterContentObserver(mediaObserver);
+
+        Log.i(DEBUG_TAG, "Stopped watchting for new media content.");
+    }
+
+    private class MediaObserver extends ContentObserver {
+        public MediaObserver() {
+            super(null);
+        }
+
+        @Override
+        public void onChange(boolean selfChange) {
```

Comment:
From the official doc, we know that this method signature is provided for compatibility with platforms, so do you think we should add another method like below.
```
/*
         * Define a method that's called when data in the
         * observed content provider changes.
         */
        @Override
        public void onChange(boolean selfChange, Uri changeUri) {
            /*
             * Ask the framework to run your sync adapter.
             * To maintain backward compatibility, assume that
             * changeUri is null.
            ContentResolver.requestSync(ACCOUNT, AUTHORITY, null);
        }
```